### PR TITLE
Update teleport-operator docs with resource naming restrictions

### PR DESCRIPTION
**Pull request by pschisa**
opening a new PR as the doc moved for adding docs update on operator naming restrictions

### CODE REVIEW

File: docs/pages/management/dynamic-resources/teleport-operator.mdx [hunk 1]
```diff
@@ -131,6 +131,12 @@ spec:
   roles: ['myrole']
 ```
 
+<Admonition type="note">
+   Kubernetes enforces all custom resource names to follow RFC 1123 which includes specifications for hostnames.
+   This requires the metadata.name of Teleport resources controlled by the operator consist of lowercase alphanumeric
```
Comment (ptgott) [inline docs/pages/management/dynamic-resources/teleport-operator.mdx:136]:
```suggestion
   This requires the `metadata.name` field of Teleport resources controlled by the operator to consist of lowercase alphanumeric
```

File: docs/pages/management/dynamic-resources/teleport-operator.mdx [hunk 2]
```diff
@@ -131,6 +131,12 @@ spec:
...
+   characters, '-' or '.', and must start and end with an alphanumeric character.
```
Comment (ptgott) [inline docs/pages/management/dynamic-resources/teleport-operator.mdx:137]:
```suggestion
   characters, `-` or `.`, and to start and end with an alphanumeric character.
```

File: docs/pages/management/dynamic-resources/teleport-operator.mdx [hunk 3]
```diff
@@ -261,6 +267,21 @@ has been locked due to a certificate generation mismatch, possibly indicating a
 In order to resolve this issue, remove any Teleport Kubernetes Operator sidecars connecting to the shared backend that are not in the same
 Kubernetes cluster or namespace.  Then redeploy the Auth Server pods.  When the operator starts, it deletes the cached information and recreates all required resources.
 
+### Teleport resource name is invalid with the Operator
+
+Kubernetes enforces all custom resource names to follow RFC 1123 which includes specifications for hostnames.  If attempting to create a resource using the
```
Comment (ptgott) [inline docs/pages/management/dynamic-resources/teleport-operator.mdx:272]:
```suggestion
Kubernetes validates all custom resource names to follow RFC 1123, which includes specifications for hostnames. If attempting to create a resource using the 
```

File: docs/pages/management/dynamic-resources/teleport-operator.mdx [hunk 4]
```diff
@@ -131,6 +131,12 @@ spec:
   roles: ['myrole']
 ```
 
+<Admonition type="note">
+   Kubernetes enforces all custom resource names to follow RFC 1123 which includes specifications for hostnames.
```
Comment (ptgott) [inline docs/pages/management/dynamic-resources/teleport-operator.mdx:135]:
```suggestion
   Kubernetes validates all custom resource names to follow RFC 1123, which includes specifications for hostnames.
```

File: docs/pages/management/dynamic-resources/teleport-operator.mdx [hunk 5]
```diff
@@ -261,6 +267,21 @@ has been locked due to a certificate generation mismatch, possibly indicating a
...
+
+```text
+The TeleportRole "my_role" is invalid: metadata.name: Invalid value: "my_role": a lowercase RFC 1123 subdomain
+must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character
+(e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
+```
+
+You will have to update the Teleport resource name to match the RFC 1123 requirements to resolve this issue.
+The metadata.name of Teleport resources controlled by the operator must consist of lowercase alphanumeric
+characters, '-' or '.', and must start and end with an alphanumeric character.
```
Comment (ptgott) [inline docs/pages/management/dynamic-resources/teleport-operator.mdx:283]:
```suggestion
The `metadata.name` of Teleport resources controlled by the operator must consist of lowercase alphanumeric
characters, `-` or `.`, and must start and end with an alphanumeric character.
```

**Review by ptgott — APPROVED** [review]
Approved with suggestions

**Review by hugoShaka — APPROVED** [review]

**Comment by pschisa** [comment]
looks like I missed the window to merge and the operator docs have been split further.  Will need to review how the docs have been split to re-find where this information is appropriate and place in new files.  Closing this one for now

**Pull request closed without merging**